src/rules/effects/prefer-effect-callback-in-block-statement.ts

```diff
@@ -26,19 +26,52 @@ export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
   },
   defaultOptions: [],
   create: (context) => {
+    const sourceCode = context.getSourceCode()
+    const nonParametrizedEffect =
+      `${createEffectExpression} > ArrowFunctionExpression > .body[type!=/^(ArrowFunctionExpression|BlockStatement)$/]` as const
+    const parametrizedEffect =
+      `${createEffectExpression} > ArrowFunctionExpression > ArrowFunctionExpression > .body[type!='BlockStatement']` as const
+    const parametrizedEffectWithinBlockStatement =
```

Comment:
I've tried to be as explicit as possible in the variables, but feel free to suggest if that's the case.